### PR TITLE
internal: add RECOMMENDATIONS_PROXY to the configuration

### PR DESCRIPTION
**Pull request by mgold1234**
this commit add RECOMMENDATIONS_PROXY to image builder configuration to support the vpn problem

### CODE REVIEW

**Review by croissanne — DISMISSED** [review]
Thanks!

File: internal/clients/recommendations/client.go [hunk 1]
```diff
@@ -20,11 +20,13 @@ type RecommendationsClient struct {
 	URL     string
 	tokener oauth2.Tokener
 	client  *http.Client
+	PROXY   string
 }
 
 type RecommendationsClientConfig struct {
 	URL     string
 	CA      string
+	PROXY   string
```
Comment (croissanne) [inline internal/clients/recommendations/client.go:29]: The naming convention should make it `Proxy` here.

**Review by croissanne — CHANGES_REQUESTED** [review]

**Comment by croissanne** [comment]
You'll need to add the proxy to the client:

https://pkg.go.dev/net/http#Client
https://pkg.go.dev/net/http#Transport
https://pkg.go.dev/net/http#ProxyURL

File: internal/clients/recommendations/client.go [hunk 2]
```diff
@@ -34,40 +37,49 @@ func NewClient(conf RecommendationsClientConfig) (*RecommendationsClient, error)
...
+		transport = &http.Transport{
+			TLSClientConfig: tlsConfig,
+		}
+	} else {
+		transport = &http.Transport{}
 	}
 
-	transport := &http.Transport{TLSClientConfig: tlsConfig}
+	if proxyURL != "" {
+		proxyURLParsed := url.URL{Host: proxyURL, Scheme: "http"}
```
Comment (croissanne) [inline internal/clients/recommendations/client.go:80]: Why not use https://pkg.go.dev/net/url#Parse?

File: internal/v1/server_test.go [hunk 3]
```diff
@@ -149,6 +150,7 @@ func startServer(t *testing.T, tscc *testServerClientsConf, conf *ServerConfig)
 	}
 	recommendClient, err := recommendations.NewClient(recommendations.RecommendationsClientConfig{
 		URL:     tscc.RecommendURL,
+		PROXY:   tscc.PROXY,
```
Comment (croissanne) [inline internal/v1/server_test.go:153]: `PROXY` -> `Proxy`

File: internal/v1/server_test.go [hunk 4]
```diff
@@ -100,6 +100,7 @@ type testServerClientsConf struct {
 	CSURL        string
 	RecommendURL string
 	OAuthURL     string
+	PROXY        string
```
Comment (croissanne) [inline internal/v1/server_test.go:103]: `PROXY` -> `Proxy`

**Review by croissanne — CHANGES_REQUESTED** [review]
Thanks, some small comments/question.

File: internal/clients/recommendations/client.go (hunk 2 above)
Comment (mgold1234) [reply internal/clients/recommendations/client.go:80]: @croissanne I tried to use proxyURLParsed, err := url.Parse(proxyURL), but it is not working, 
beucase we didnt define the http:// for proxy paramemter at Vault system.

Comment (croissanne) [reply internal/clients/recommendations/client.go:80]: ah well I'll update it to include the scheme then.

Comment (croissanne) [reply internal/clients/recommendations/client.go:80]: updated!

File: internal/v1/server_test.go (hunk 4 above)
Comment (croissanne) [reply internal/v1/server_test.go:103]: This is still valid, naming convention

**Review by croissanne — COMMENTED** [review]
Thank you, 1 more comment

**Review by croissanne — APPROVED** [review]
Perfect, thank you!

**Pull request merged**